library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java

```diff
@@ -227,6 +227,7 @@ private void shutDown() {
         Log.d("Shutting down service");
         getContentResolver().unregisterContentObserver(downloadManagerContentObserver);
         downloadScanner.shutdown();
+        executor.shutdownNow();
```

Comment:
This is weird because the executor shouldn't have any pending threads as they should imply that there is an active batch and while an active batch the DownloadService should remain active.

When no downloads can proceed, all threads should die, the status of the batch should not be active and the service  die nicely, this shouldn't be necessary.

Maybe there is a bug somewhere else that is preventing the work that I've described to happen?